src/azure-cli/azure/cli/command_modules/acs/_validators.py

```diff
@@ -259,3 +259,75 @@ def validate_vnet_subnet_id(namespace):
         from msrestazure.tools import is_valid_resource_id
         if not is_valid_resource_id(namespace.vnet_subnet_id):
             raise CLIError("--vnet-subnet-id is not a valid Azure resource ID.")
+
+
+def validate_nodepool_labels(namespace):
+    """Validates that provided node labels is a valid format"""
+
+    if hasattr(namespace, 'nodepool_labels'):
+        labels = namespace.nodepool_labels
+    else:
+        labels = namespace.labels
+
+    if labels is None:
+        return
+
+    if isinstance(labels, list):
+        labels_dict = {}
+        for item in labels:
+            labels_dict.update(validate_label(item))
+        after_validation_labels = labels_dict
+    else:
+        after_validation_labels = validate_label(labels)
+
+    if hasattr(namespace, 'nodepool_labels'):
+        namespace.nodepool_labels = after_validation_labels
+    else:
+        namespace.labels = after_validation_labels
+
+
+def validate_label(label):
+    """Validates that provided label is a valid format"""
+    prefix_regex = re.compile(r"^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$")
+    name_regex = re.compile(r"^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$")
+    value_regex = re.compile(r"^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$")
+
+    if label == "":
+        return {}
```

Comment:
Do you want to support clearing labels with `--labels ""`? Just like `tags`?

Reply:
We will in the future. But for short to mid term, we don't allow labels update after creation.